tests/src/JIT/SIMD/Vector3TestNative.cpp

```diff
@@ -0,0 +1,161 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <stdio.h>
+
+#if defined(__clang__)
+#define EXPORT(type) __attribute__((visibility("default"))) extern "C" type
+#elif defined(_MSC_VER)
+#define EXPORT(type) extern "C" __declspec(dllexport) type
+#else // defined(__clang__)
+#define EXPORT(type) type
+#endif // !defined(__clang__)
+
+#if !defined(_MSC_VER)
+#if __i386__
+#define __stdcall __attribute__((stdcall))
+#else // __i386__
+#define __stdcall
+#endif  // !__i386__
+#endif // !defined(_MSC_VER)
+
+typedef struct _Vector3 
+{
+    float x;
+    float y;
+    float z;    
+} Vector3;
+
+typedef struct _DT 
+{
+   Vector3 a;
+   Vector3 b;
+} DT;
+
+//
+// PInvoke native call for Vector3 size check
+//
+
+EXPORT(int) __stdcall nativeCall_PInvoke_CheckVector3Size() 
+{
+    printf("nativeCall_PInvoke_CheckVector3Size: sizeof(Vector3) == %d\n", (int)sizeof(Vector3));
```

Comment:
In my experience, the native console output is not synchronized with the managed output, and I needed to insert fflush(stdout) to have things better interleaved.  I could have been doing something wrong, but I thought I would mention it in case it's useful.